bundles/org.eclipse.ui/src/org/eclipse/ui/internal/SearchDecoration.java

```diff
@@ -62,21 +62,55 @@ public static boolean validateRegex(String regex, ControlDecoration targetDecora
 	 * @return The appropriate error message if the regex is invalid or an empty
 	 *         string if the regex is valid.
 	 */
-	private static String getValidationError(String regex) {
+	private static String getValidationError(String regex, org.eclipse.swt.widgets.Control targetControl) {
+
 		try {
 			Pattern.compile(regex);
 			return ""; //$NON-NLS-1$
+
 		} catch (PatternSyntaxException e) {
-			String message = e.getLocalizedMessage();
+			return buildValidationErrorString(e, targetControl);
+		}
+	}
+
+	private static String buildValidationErrorString(PatternSyntaxException e, org.eclipse.swt.widgets.Control targetControl) {
+		GC gc = new GC(targetControl);
+
+		String description = e.getDescription();
+		int errorIndex = e.getIndex();
+		String pattern = e.getPattern();
+
+		if (errorIndex == -1) {
+			return description;
+		}
```

Comment:
The GC and the pattern are needed later, no need for them if you do an early return.

```suggestion
		String description = e.getDescription();
		int errorIndex = e.getIndex();

		if (errorIndex == -1) {
			return description;
		}
		
		GC gc = new GC(targetControl);
		String pattern = e.getPattern();
```

Reply:
done